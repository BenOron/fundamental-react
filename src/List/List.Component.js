import path from 'path';
import React from 'react';
import { Button, List } from '..';
import { ComponentPage, Example } from '../_playground';

export const ListComponent = () => {
    return (
        <ComponentPage
            description={`Lists and tables are similar as both usually contain a vertical list of data,
                but lists generally contain basic data and tables tend to hold more complex data.
                If the list is a complex hierarchy, it is best to use a **Tree**.`}
            sourceModulePath={path.join(__dirname, './List')}
            title='List Group'>

            <Example
                description='A link can be used to allow the user to access more details about the item.'
                title='Simple List'>
                <List>
                    <List.Item><a style={{ cursor: 'pointer' }}>List item 1</a></List.Item>
                    <List.Item>List item 2</List.Item>
                    <List.Item><a style={{ cursor: 'pointer' }}>List item 3</a></List.Item>
                    <List.Item>List item 4</List.Item>
                </List>
            </Example>

            <Example
                description='The List item can contain quick actions.'
                title='List with Actions'>
                <List>
                    <List.Item>List item 1
                        <List.ItemActions>
                            <Button glyph='edit' option='light' />
                        </List.ItemActions>
                    </List.Item>
                    <List.Item>List item 2
                        <List.ItemActions>
                            <Button glyph='edit' option='light' />
                        </List.ItemActions>
                    </List.Item>
                    <List.Item>List item 3
                        <List.ItemActions>
                            <Button glyph='edit' option='light' />
                        </List.ItemActions>
                    </List.Item>
                    <List.Item>List item 4
                        <List.ItemActions>
                            <Button glyph='edit' option='light' />
                        </List.ItemActions>
                    </List.Item>
                </List>
            </Example>

            <Example
                description='Checkboxes can be include on the left of each line for such purposes as bulk actions.'
                title='List with Checkboxes'>
                <List>
                    <List.Item><List.ItemCheckbox>List item 1</List.ItemCheckbox></List.Item>
                    <List.Item><List.ItemCheckbox>List item 2</List.ItemCheckbox></List.Item>
                    <List.Item><List.ItemCheckbox>List item 3</List.ItemCheckbox></List.Item>
                    <List.Item><List.ItemCheckbox>List item 4</List.ItemCheckbox></List.Item>
                </List>
            </Example>
        </ComponentPage>
    );
};
