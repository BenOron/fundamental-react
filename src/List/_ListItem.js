import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { useState } from 'react';

const ListItem = ({
    className,
    children,
    selected,
    ...props
}) => {

    let { isSelected, setIsSelected } = useState(selected);

    const ListItemClasses = classnames(
        'fd-list__item',
        className
    );

    return (
        <li
            {...props}
            aria-selected={isSelected}
            className={ListItemClasses}
            onClick={() => setIsSelected(!isSelected)}
            role='option'
            tabIndex='0'>
            {children}
        </li>

    );
};

ListItem.displayName = 'List.Item';

ListItem.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    selected: PropTypes.bool
};

ListItem.propDescriptions = {
    selected: 'Set to **true** to set state of the list item to "selected".'
};

export default ListItem;
