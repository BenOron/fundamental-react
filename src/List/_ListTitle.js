import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const ListTitle = ({
    className,
    children,
    ...props
}) => {
    const listTitleClasses = classnames(
        'fd-list__title',
        className
    );

    return (
        <span {...props} className={listTitleClasses}>
            {children}
        </span>

    );
};

ListTitle.displayName = 'List.Title';

ListTitle.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default ListTitle;
