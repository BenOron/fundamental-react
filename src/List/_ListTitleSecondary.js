import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const ListTitleSecondary = ({
    className,
    children,
    ...props
}) => {
    const listTitleSecondaryClasses = classnames(
        'fd-list__secondary',
        className
    );

    return (
        <span {...props} className={listTitleSecondaryClasses}>
            {children}
        </span>

    );
};

ListTitleSecondary.displayName = 'List.TitleSecondary';

ListTitleSecondary.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
};

export default ListTitleSecondary;
