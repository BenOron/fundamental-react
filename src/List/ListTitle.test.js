import List from './List';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<ListTitle />', () => {
    const ListTitle = (
        <List.Title>
            List item 1
        </List.Title>
    );

    test('create list Title item', () => {
        let component = renderer.create(ListTitle);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the ListTitle component', () => {
            const element = mount(<List.Title data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
