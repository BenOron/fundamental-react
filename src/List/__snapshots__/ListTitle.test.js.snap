// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListTitle /> create list Title item 1`] = `
<span
  className="fd-list__title"
>
  List item 1
</span>
`;
